### ISSUE TITLE
[3.0.0-beta1]Table is not registered with a Database. Did you forget the @Table annotation?

### ISSUE THREAD

**Issue by KChernenko** [Raizlabs/DBFlow#515]
But I have the folowing structure:

```Java
@Database(name = WeatherDB.NAME, version = WeatherDB.VERSION)
public class WeatherDB {

    public static final String NAME = "Weather";

    public static final int VERSION = 1;
}
```
And table:
``` Java
@Table(database = WeatherDB.class)
public class WeatherTable extends BaseModel {

    @Column
    @PrimaryKey
    private long dateStamp;
    @Column
    private double dayTemperature;
    @Column
    private double minTemperature;
    @Column
    private double maxTemperature;
    @Column
    private double nightTemperature;
    @Column
    private double eveTemperature;
    @Column
    private double mornTemperature;
    @Column
    private int humidity;
    @Column
    private String description;
    @Column
    private double windSpeed;

   //getters and setters
}
```

Full stacktrace:
```
Process: me.bitfrom.weatherapp, PID: 3939
                                                                     com.raizlabs.android.dbflow.structure.InvalidDBConfiguration: Table: me.bitfrom.weatherapp.database.WeatherTable is not registered with a Database. Did you forget the @Table annotation?
                                                                         at com.raizlabs.android.dbflow.config.FlowManager.getDatabaseForTable(FlowManager.java:109)
                                                                         at com.raizlabs.android.dbflow.config.FlowManager.getModelAdapter(FlowManager.java:271)
                                                                         at com.raizlabs.android.dbflow.config.FlowManager.getTableName(FlowManager.java:68)
                                                                         at com.raizlabs.android.dbflow.sql.language.From.<init>(From.java:55)
                                                                         at com.raizlabs.android.dbflow.sql.language.Select.from(Select.java:58)
                                                                         at me.bitfrom.weatherapp.ui.fragments.TodaysWeatherFragment.onViewCreated(TodaysWeatherFragment.java:31)
                                                                         at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:908)
                                                                         at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
                                                                         at android.app.BackStackRecord.run(BackStackRecord.java:834)
                                                                         at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1452)
                                                                         at android.app.FragmentManagerImpl$1.run(FragmentManager.java:447)
                                                                         at android.os.Handler.handleCallback(Handler.java:739)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                         at android.os.Looper.loop(Looper.java:211)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5389)
                                                                         at java.lang.reflect.Method.invoke(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:372)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1020)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:815)
```

**Comment by MaciejCzekanski** [comment]
I have the same problem

**Comment by KChernenko** [comment]
@MaciejCzekanski are you using Dagger 2 in your project?

**Comment by MaciejCzekanski** [comment]
@KChernenko Yes, I can't make them both work at the same time. Do you have any solution?

**Comment by simplysoft** [comment]
Are you using proguard? If so, this config might help (solved the issue in our case)

```
-keep class * extends com.raizlabs.android.dbflow.config.DatabaseHolder { *; }
```

**Comment by KChernenko** [comment]
@simplysoft unfortunatelly proguard is disabled. I think it doesn't work in our case (me and @MaciejCzekanski) because of Dagger 2.

**Comment by MaciejCzekanski** [comment]
Yes, I don't use Proguard. I'm not sure what I've changed, but now I get compile-time error, that Dagger's Component class is missing. When I remove @Table annotation then it compiles, but obviously crashes at runtime with error "Table is not registered with a Database".

**Comment by KChernenko** [comment]
@MaciejCzekanski have the same problem. Lets mark @agrosner here to infrom.

**Comment by agrosner** [comment]
There are issues with annotation processors and how order can matter. Dagger2 has some issues with that, and unfortunately I do not know of a solution of how to fix that

**Comment by amugox** [comment]
This same error is almost making me now think of an alternative ORM

**Comment by amugox** [comment]
Finally it works for me! I had forgotten to add the Application class on the manifest

<application
        android:name=".App.TotoApp":

**Comment by MaciejCzekanski** [comment]
@amugox I have Application class referenced in manifest and it doesn't work. Do you also use Dagger 2?

**Comment by epool** [comment]
I am getting same error using Dagger 2 =/

**Comment by Ekt0s** [comment]
I'm getting the same error but just on one of my project, with the exact same database configuration, the one working is the [android-testing project](https://github.com/googlesamples/android-testing)

**Comment by epool** [comment]
A workaround I am using to face this in my development process is doing `Build -> Build Project` and then run it again. =/

**Comment by agrosner** [comment]
are you guys using 2.0.0+ of the Gradle plugin with instant run enabled?

**Comment by epool** [comment]
Yes, I am using `classpath 'com.android.tools.build:gradle:2.0.0-alpha7'`

**Comment by agrosner** [comment]
Im 95% sure its an instant run bug.

**Comment by rogerhu** [comment]
+1 ProGuard rule fixed things for me.  I suggest it be added a consumer proguard rule for the library?
 
```-keep class * extends com.raizlabs.android.dbflow.config.DatabaseHolder { *; }```
```

**Comment by jesdiart** [comment]
Hi all

I am having the same problem here, in intrumentated unit tests. I am using Dagger 1.2.2 and the build tools are version 1.5.0.

I have an abstract test case that injects the dependencias using dagger:

`
public abstract class FlowTestCase extends InstrumentationTestCase {

    ObjectGraph objectGraph;

    public FlowTestCase() {
        FlowLog.setMinimumLoggingLevel(FlowLog.Level.I);
    }

    protected void setUp(){
        objectGraph = ObjectGraph.create(new AndroidTestModule(getInstrumentation().getContext())).plus(getModules().toArray());
        objectGraph.inject(this);
    }
`

But I get the error while trying to delete the tables in the setUp in each test:

`
@RunWith(AndroidJUnit4.class)
public class CredentialTypesTests extends FlowTestCase {

    @Inject
    LocalDataManager localDataManager;

    @Override
    protected List<Object> getModules() {
        return new ArrayList<>();
    }

    @Before
    public void setUp() {
        super.setUp();
        Delete.tables(MyTable.class);
    }

    @After
    public void tearDown() {
        Delete.tables(MyTable.class);
    }
`

**Comment by viethoang88** [comment]
I can confirm that this issue is caused by instant run. I disabled it and it fixed the issue.

**Comment by chandramohan11mx08** [comment]
I can confirm that this issue is caused by instant run. I disabled it and it fixed the issue.

**Comment by chandramohan11mx08** [comment]
It happens for me. I am not using dagger, proguard is disabled. I am using android annotations and gradle com.android.tools.build:gradle:2.0.0-alpha7.

It is works well when project is rebuild every time before run.

**Comment by agrosner** [comment]
use 2.0.0-beta2 - they explicitly fixed the issue:
http://tools.android.com/recent/androidstudio20beta2availableincanarychannel
```
Dagger2 and other annotation-processor based libraries and plugins: In 2.x we've been using the new "incremental Java compilation" feature in Gradle, but this feature does not work well with annotation processor. In beta2, we automatically turn off incremental compilation if a project appears to be using annotation processors through the "android-apt" plugin. Manually wired processors will still fail (see https://github.com/google/dagger/issues/298).
```

**Comment by prashantwosti** [comment]
Unfortunately, this issue persists on  2.0  Beta2 (143.2586769).

**Comment by agrosner** [comment]
there are a few potential factors associated with why it fails. 1 is gradle plugin for Android. What version are you using of the Android gradle plugin?

**Comment by iam1492** [comment]
still exists in 2.0 Beta 4. Issue disappeared when disable Instant Run.

**Comment by iam1492** [comment]
still not fixed 2.0 Beta 5.

**Comment by garmax1** [comment]
Seems was fixed in com.android.tools.build:gradle:2.0.0-beta6

**Comment by thekalinga** [comment]
Any idea on whether they disabled instant run for any non-compatible apt they find/instant run works seamlessly with dbflow

**Comment by xgbxmgg** [comment]
still not fixed 2.0 Beta 5.

**Comment by levibostian** [comment]
Bummer. Instant run fixed the issue for me as well. Too bad as instant run is pretty sweet.

**Comment by TeddyGammel** [comment]
I am seeing this issue in dbflow version 3.1.1. Disabled instant run on my studio the gradle version being used is 2.1.3. Even with 2.0.0-beta6, its not working.

**Comment by TeddyGammel** [comment]
@agrosner request your help here. Thank you.

**Comment by hoanv810** [comment]
@TeddyGammel Please make sure you have initialized FlowConfig in your Application class!

**Comment by elroid** [comment]
I'm having this problem as well. DaggerApplicationComponent and DaggerConfigPersistentComponent simply aren't getting created properly when I enable DBFlow. Looks like i'll need to go with a different ORM... :-(

**Comment by agrosner** [comment]
Ensure you dont have any compile issues. also that you didnt muck up one of the other annotation processor dependant libs, as that will cause side effects in other annotation processors. Ensure incremental compilation is not interfering with annotation processing (it should be fixed as of 2.2.2) If your Dagger2 components are not getting generated, its likely its supressing DBFlow issues or you have an error in Dagger2.

**Comment by elroid** [comment]
Having done some more experimentation, it seems not to be DBFlow at fault, but rather android-apt. As you alluded to, as of the gradle 2.2.x plugin, this lib is deprecated ( https://bitbucket.org/hvisser/android-apt/wiki/Migration ) and dagger fails the generation step when I include it. But when I try to add dbflow's compile directives, i get: 
`Error:(151, 0) Could not find method apt() for arguments [com.github.Raizlabs.DBFlow:dbflow-processor:4.0.0-beta2] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.`
Any suggestions for working with DBFlow using the newest gradle plugin?
